### ISSUE TITLE
Support for file: and tilde-expanded homedirs

### ISSUE THREAD

**Issue by shalomb** [aelsabbahy/goss#56]
It would be nice if goss could support [tilde-expanded homedirs]( https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html) `file` tests using the form '~user/' as opposed to a a priori convention of '/home/user/' (or to be forced to do a lookup first) as the homedirs may exist on NFS under a different directory, are placed differently for system users, may change over time, etc.

goss fails these checks when you expect them to pass.

 ```
cat users.yaml
---

file:  
  ~sshd:
    exists: true
  ~systemd-timesync:
    exists: true
   ~testuser/:
     exists: true
   ~testuser/.ssh/:
     exists: true
   ~testuser/.ssh/authorized_keys:
     exists: true
   ~testuser/.ssh/known_hosts:
     exists: true
```

```
# goss --gossfile users.yaml validate </dev/null -f tap
1..6
not ok 1 - File: ~sshd: exists: doesn't match, expect: [true] found: [false]
not ok 2 - File: ~testuser/.ssh/authorized_keys: exists: doesn't match, expect: [true] found: [false]
not ok 3 - File: ~testuser/.ssh/known_hosts: exists: doesn't match, expect: [true] found: [false]
not ok 4 - File: ~systemd-timesync: exists: doesn't match, expect: [true] found: [false]
not ok 5 - File: ~testuser: exists: doesn't match, expect: [true] found: [false]
not ok 6 - File: ~testuser/.ssh/: exists: doesn't match, expect: [true] found: [false]
```
 
 when this is form is typically usable otherwise
 
 ```
root@test:/etc/goss# ls -ld ~systemd-timesync ~sshd ~testuser ~testuser/.ssh ~testuser/.ssh/authorized_keys ~testuser/.ssh/known_hosts
drwxr-xr-x 15 root root 360 Mar 23 23:49 /run/systemd
drwxr-xr-x  2 root root  40 Mar 23 07:11 /var/run/sshd
 drwxr-xr-x 3 testuser testuser   99 Mar 23 23:51 /home/testuser
 drwx------ 2 testuser testuser  149 Mar 23 23:51 /home/testuser/.ssh
 -r-------- 1 testuser testuser 2812 Mar 23 23:50 /home/testuser/.ssh/authorized_keys
 -rw------- 1 testuser testuser 2184 Mar 23 23:51 /home/testuser/.ssh/known_hosts
 ```

**Comment by shalomb** [comment]
You can workaround this by using a `command` instead

```
command:
  "ls -ld ~testuser":
    exit-status: 0
  "ls -ld ~testuser/.ssh":
    exit-status: 0
  "ls -ld ~testuser/.ssh/authorized_keys":
    exit-status: 0
  "ls -ld ~testuser/.ssh/known_hosts":
    exit-status: 0
```

**Comment by aelsabbahy** [comment]
Released in [v0.1.4](https://github.com/aelsabbahy/goss/releases/tag/v0.1.4)
